### DISCUSSION TITLE
Write Mode - Upgrade undo command !

### DISCUSSION THREAD

**Discussion by zwk5131** [neovim/neovim#27161] [Ideas]
Hello 

I really like to take notes in nvim. The only problem is the undo command...
If you write a long line and want to undo something it will delete the hole line. To fix that problem there is the command ``<c-g>u``.
But you have to put it manually ... and that's not really a good workflow...

So my idea was to make a "WRITE-MODE" where you automate the command and i came up with this:
1) ``keymap("i", "<space>", "<space><c-g>u", opts)`` --undo per word
2) ``keymap("i", "<tab>", "<tab><c-g>u", opts)`` --undo per tab
3) ``keymap("i", "<bs>", "<bs><c-g>u", opts)`` --undo per char when deleted with ``<bs>``
4) ``keymap("i", "<enter>", "<enter><c-g>u", opts)`` --undo newline

It would be nice to implement this with a command like this:
**set writemode**

NOTE!!!
If you wan't to get more info about undo and this ``<c-g>u`` command go to this website: 
https://learnvim.irian.to/basics/undo

**Comment by zeertzjq** [comment]
`:h undo-break`

**Comment by zwk5131** [comment]
@zeertzjq 
Thanks for the hint but that only covers 1 of my 4 aspects...
I saw in the help info this command `:undojoin` which corresponded with number 3..
